OfflineUX.md

```diff
@@ -0,0 +1,129 @@
+#### Offline UX Patterns
+
+### Contents
+* [Motivation & Philosophy](#motivation-&-philosophy)
+* [UX Pattern Flowchart](#ux-pattern-flowchart)
+* [How to Answer Questions on the Flow Chart](#how-to-answer-questions-on-the-flowchart)
+* [Description of the Patterns](#description-of-the-patterns)
+    - [None - No Offline Behavior](#none---no-offline-behavior)
+    - [A - Optimistic Without Feedback](#a---optimistic-without-feedback)
+    - [B - Optimistic With Feedback](#b---optimistic-with-feedback)
+    - [C - Blocking Form](#c---blocking-form)
+    - [D - Full Page Blocking](#d---full-page-blocking)
+
+### Motivation & Philosophy
+
+Understanding the offline behavior of our app is vital to becoming a productive contributor to the Expensify codebase. Our mission is to support our users in every possible environment, and often our app is used in places where a stable internet connection is not guaranteed. 
+
+The most important concept to keep in mind while reading this document is that we want to allow users to do as much as possible when offline. At first, this might seem impossible because almost everything the user can touch in our app is related to an API request. However, in many cases, we can save that API request and assume it will succeed when the user is back online. We then allow the user to proceed as if their request already succeeded. We call this an optimistic response. Here, we use the word **optimistic** to indicate that we’re confident the request will succeed when the user is online. 
+
+<div style="margin: 15px; padding: 15px; border: 3px solid grey">
```

Comment:
This doesn't look like it's doing anything?

![image](https://user-images.githubusercontent.com/1228807/170135272-f434872e-1888-4346-be91-d909b78a86e1.png)

Reply:
Oh, strange. It works for me in the markdown preview in VS Code. I'll take it out if it doesn't render in GH. 

<img width="1324" alt="Screen Shot 2022-05-24 at 5 06 58 PM" src="https://user-images.githubusercontent.com/49007721/170151511-a25bd420-f7f2-4941-8ffe-9ae25809041e.png">

Reply:
Wanna just remove it now? It doesn't show up in the rich diff so it probably won't work.

Reply:
Or you could do a `<hr />` above and below it.